pkg/controllers/management/auth/crtb_handler.go

```diff
@@ -52,7 +52,8 @@ var clusterManagmentPlaneResources = map[string]string{
 	"nodepools":                   "management.cattle.io",
 	"notifiers":                   "management.cattle.io",
 	"podsecuritypolicytemplateprojectbindings": "management.cattle.io",
-	"projects": "management.cattle.io",
+	"projects":      "management.cattle.io",
+	"etcdsnapshots": "rke.cattle.io",
 }
 
 type crtbLifecycle struct {
```

Comment:
> Because of this, roles get access to individual snapshots. If no snapshots are stored, the users do not have access to view or create snapshots in the UI.

Can you clarify what you mean by "roles get access to snapshots" there ? 

The idea that you grant them this resource in the cluster-specific namespace makes sense. But earlier you said backups are actually available in the fleet-default NS, so does the backup never live in the cluster namespace ? And therefore this is sort of a "fake it" way to get the schema ?

Reply:
You are correct. After a snapshot is created, it is stored in the fleet-default namespace in the local cluster, then the specific resource is added to the role that is bound to cluster-owners. Backups will not live in the cluster namespace, but it allows the schema to know they have access to the CRD `etcdsnapshots.rke.cattle.io`. This lets the UI know the given user has access to create new snapshots when none are available.

In the future, a better solution would be to add the snapshots to the cluster namespace instead of fleet-default, but the current behavior is needed for provisioning with fleet.

Reply:
These objects should always live in the namespace that correlates to the provisioning cluster. We have an issue filed to handle this here: https://github.com/rancher/dashboard/issues/5295